src/components/select/select.ts

```diff
@@ -0,0 +1,377 @@
+    if (this.invalid) this.focus();
+    return !this.invalid;
+  }
+
+  public override async firstUpdated() {
+    super.firstUpdated();
+    await this.updateComplete;
+
+    const selectedItem = this.allItems.find((i) => i.selected) ?? null;
+    if (selectedItem) this.value = selectedItem.value;
+    else this.updateSelected();
+  }
+
+  @watch('selectedItem')
+  protected updateValue() {
+    this.value = this.selectedItem?.value ?? this.value;
+  }
+
+  @watch('value')
+  protected updateSelected() {
+    if (this.allItems.length === 0) return;
+
+    if (this.selectedItem?.value !== this.value) {
+      const matches = this.allItems.filter((i) => i.value === this.value);
+      const index = this.allItems.indexOf(matches[matches.length - 1]);
+
+      if (index === -1) {
+        this.value = undefined;
+        this.clearSelection();
+        return;
+      }
+
+      this.select(index);
+    }
+  }
+
+  @watch('value')
+  protected validate() {
+    this.updateComplete.then(() => this.reportValidity());
+  }
+
+  protected selectNext() {
+    const activeItemIndex = [...this.allItems].indexOf(
+      this.selectedItem ?? this.activeItem
+    );
+
+    const next = this.getNearestSiblingFocusableItemIndex(
+      activeItemIndex ?? -1,
+      1
+    );
+    this.selectItem(this.allItems[next], true);
+  }
+
+  protected selectPrev() {
+    const activeItemIndex = [...this.allItems].indexOf(
+      this.selectedItem ?? this.activeItem
+    );
+    const prev = this.getNearestSiblingFocusableItemIndex(
+      activeItemIndex ?? -1,
+      -1
+    );
+    this.selectItem(this.allItems[prev], true);
+  }
+
+  protected selectFirstActiveItem() {
+    const items = this.allItems.filter((i) => !i.disabled);
+    const index = this.allItems.indexOf(items[0]);
+    const item = this.allItems[index];
+
+    if (item.value !== this.value) {
+      this.selectItem(item, true);
+    }
+  }
+
+  protected selectLastActiveItem() {
+    const items = this.allItems.filter((i) => !i.disabled);
+    const index = this.allItems.indexOf(items[items.length - 1]);
+    const item = this.allItems[index];
+
+    if (item.value !== this.value) {
+      this.selectItem(item, true);
+    }
+  }
+
+  protected selectItemByKey(event: KeyboardEvent): void {
+    if (
+      !event ||
+      !event.key ||
+      event.key.length > 1 ||
+      event.key === ' ' ||
+      event.key === 'spacebar'
+    ) {
+      // ignore longer keys ('Alt', 'ArrowDown', etc) AND spacebar (used of open/close)
+      return;
+    }
+
+    const currentTime = Date.now();
+
+    if (currentTime - this.lastKeyTime > 500) {
+      this.searchTerm = '';
+    }
+
+    this.searchTerm += event.key;
+    this.lastKeyTime = currentTime;
+
+    const item = this.allItems
+      .filter((i) => !i.disabled)
+      .find((i) =>
+        i.textContent
+          ?.trim()
+          .toLowerCase()
+          .startsWith(this.searchTerm.toLowerCase())
+      );
+    if (item) this.selectItem(item);
```

Comment:
I believe this is intended to be equivalent with other keyboard navigation - i.e. activate, rather than select. I realize that's how the native select does it, but it also doesn't have a separate active state and does the same with arrows.
Might need a spec update